internal/provider/network_external_resource.go

```diff
@@ -0,0 +1,314 @@
+						Required:            true,
+						MarkdownDescription: "Service account email",
+						PlanModifiers: []planmodifier.String{
+							stringplanmodifier.RequiresReplace(),
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func (r *ExternalNetworkResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data *ExternalNetworkResourceModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	importReq := &network.ImportNetworkRequest{
+		Name:        data.Name.ValueString(),
+		ProjectId:   data.ProjectID.ValueString(),
+		Description: data.Description.ValueString(),
+	}
+
+	switch {
+	case data.AWS != nil:
+		importReq.Params = &network.ImportNetworkRequest_Aws{
+			Aws: &network.ImportAWSVPCRequest{
+				VpcId:      data.AWS.VPCID.ValueString(),
+				RegionId:   data.AWS.RegionID.ValueString(),
+				AccountId:  data.AWS.AccountID.ValueString(),
+				IamRoleArn: data.AWS.IAMRoleARN.ValueString(),
+			},
+		}
+	case data.Google != nil:
+		// TODO export google to public API
+	}
+
+	opObj, err := r.networkService.Import(ctx, importReq)
+	if err != nil {
+		resp.Diagnostics.AddError("failed to create", err.Error())
+		return
+	}
+	op, err := r.sdk.WrapOperation(opObj, err)
+	if err != nil {
+		resp.Diagnostics.AddError("failed to create", err.Error())
+		return
+	}
+	err = op.Wait(ctx)
+	if err != nil {
+		resp.Diagnostics.AddError("failed to create", err.Error())
+	}
+
+	data.ID = types.StringValue(op.ResourceId())
+
+	net, err := r.networkService.Get(ctx, &network.GetNetworkRequest{NetworkId: op.ResourceId()})
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to get network", fmt.Sprintf("failed request, error: %v", err))
+		return
+	}
+	data.Ipv4CidrBlock = types.StringValue(net.Ipv4CidrBlock)
+	data.Ipv6CidrBlock = types.StringValue(net.Ipv6CidrBlock)
+
+	// Save data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *ExternalNetworkResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var data *ExternalNetworkResourceModel
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	net, err := r.networkService.Get(ctx, &network.GetNetworkRequest{NetworkId: data.ID.ValueString()})
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to get network", fmt.Sprintf("failed request, error: %v", err))
+		return
+	}
+
+	data.ID = types.StringValue(net.Id)
+	data.Name = types.StringValue(net.Name)
+	data.ProjectID = types.StringValue(net.ProjectId)
+	data.Description = types.StringValue(net.Description)
+	data.Ipv4CidrBlock = types.StringValue(net.Ipv4CidrBlock)
+	data.Ipv6CidrBlock = types.StringValue(net.Ipv6CidrBlock)
+
+	switch er := net.ExternalResources.(type) {
```

Comment:
What if `net.ExternalResources` is `nil`?

Reply:
then neither `aws` nor `google` fields will be filled

Reply:
added more tests for that